feature/platform/transceiver/tests/zr_temperature_test/zr_temperature_test.go

```diff
@@ -92,15 +92,18 @@ func TestZRTemperatureState(t *testing.T) {
 		t.Fatalf("%s Transceiver is not 400ZR its of type: %v", transceiverName, dp1.PMD())
 	}
 	component1 := gnmi.OC().Component(transceiverName)
-	subcomponents := gnmi.LookupAll[*oc.Component_Subcomponent](t, dut1, component1.SubcomponentAny().State())
-	for _, s := range subcomponents {
-		subc, ok := s.Val()
-		if ok {
-			sensorComponent := gnmi.Get[*oc.Component](t, dut1, gnmi.OC().Component(subc.GetName()).State())
-			if sensorComponent.GetType() == sensorType {
-				scomponent := gnmi.OC().Component(sensorComponent.GetName())
-				if scomponent != nil {
-					component1 = scomponent
+	// subcomponents check not required for CISCO
+	if dut1.Vendor() != ondatra.CISCO {
```

Comment:
can we rewrite this a little for readability:

var compWithTemperature *oc.Component
switch dut1.Vendor() {
case ondatra.CISCO:
  compWithTemperature = gnmi.OC().Component(transceiverName)
default:
  <...add the for loop logic here...>
}

Reply:
Also note, the canonical OC to get temperature is through a subcomponent sensor.  If cisco does not support this, then a deviation should be defined and used instead of `if dut1.Vendor() != ondatra.CISCO `

Reply:
Ack - I'll push these changes for zr_temperature_test in my next PR.